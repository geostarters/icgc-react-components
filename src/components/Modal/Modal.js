//@flow

import React from "react";
import PropTypes from "prop-types";
import { Modal as ModalSUI, Button, Icon, Image } from "semantic-ui-react";

export default function Modal(props) {

	return (
		<ModalSUI {...props}>
			{props.headertitle && <ModalSUI.Header className={props.headerClassName} content={props.headertitle}></ModalSUI.Header>}
			<ModalSUI.Content className={props.contentClassName}>
				{props.contentImage &&  <Image wrapped size='medium' src={props.contentImage} />}
				{props.children}
			</ModalSUI.Content>
			{props.actionbuttons && <ModalSUI.Actions className={props.actionsClassName}>
				{props.actionbuttons.map((btn, index) => <Button primary={false} key={index} color={btn.color} onClick={btn.handler}>
					{btn.icon && <Icon name={btn.icon}></Icon>} {btn.label}
				</Button>
				)}
			</ModalSUI.Actions>
			}
		</ModalSUI>
	);

}

Modal.propTypes = {
	size: PropTypes.string,
	trigger: PropTypes.object,
	children: PropTypes.object,
	actionbuttons: PropTypes.array,
	open: PropTypes.bool,
	closeIcon: PropTypes.bool,
	headertitle: PropTypes.string,
	contentImage: PropTypes.string,
	actionsClassName: PropTypes.string,
	headerClassName: PropTypes.string,
	contentClassName: PropTypes.string,
};

Modal.defaultProps = {
	closeIcon: true,
	headertitle: null,
	actionbuttons: null,
	trigger: null
};
