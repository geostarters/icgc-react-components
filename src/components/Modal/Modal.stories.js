//@flow

import React from "react";

import { action } from "@storybook/addon-actions";

import Button from "../Button/Button";

import Modal from "./Modal";

import "semantic-ui-css/semantic.min.css";

export default {
	title: "Modal",
	component: Modal
};

const Template = args => (<Modal {...args} trigger={<Button>Show Modal</Button>}>text content</Modal>);

export const _Modal = Template.bind({});
_Modal.storyName = "Modal";

export const ModalWithHeader = Template.bind({});
ModalWithHeader.storyName = "Modal with header";
ModalWithHeader.args = {
	headertitle: "text header"
};

export const ModalWithButtons = args => (<Modal {...args} trigger={<Button>Show Modal</Button>}>
	<div>
		<p>Aquest estil no ha estat generat a Contextmaps i per tant no es podrà editar.</p>
		<p>Vols continuar amb la càrrega?</p>
	</div>
</Modal>);
ModalWithButtons.storyName = "Modal with buttons";
ModalWithButtons.args = {
	headertitle: "Avís",
	actionbuttons: [
		{label: "Carregar", color: "green", handler: action("onClick"), icon: "check" },
		{label: "Cancel·lar", color: "grey", handler: action("onClick"), icon: "remove"}
	]
};
