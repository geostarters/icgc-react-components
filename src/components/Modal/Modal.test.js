// @flow

import React from "react";
import { Button } from "semantic-ui-react";
import Modal from "./Modal";
import  { shallow, mount } from "enzyme";
import toJSON from "enzyme-to-json";

describe("Modal Component", () => {

	const mockProps = {
		headertitle: "Hola",
		open: true,
		actionbuttons: [
			{label: "No", color: "red", handler: jest.fn(), icon: "remove"},
			{label: "SÃ­", color: "green", handler: jest.fn(), icon: "check" }
		]
	};

	const component = shallow(<Modal {...mockProps}></Modal>);

	it("renders correct properties", () => {

		expect(toJSON(component)).toMatchSnapshot();

	});

	it("renders with a class named modals", () => {

		const mockProps = {
			open: true,
		};
		const component = mount(<Modal {...mockProps}></Modal>);
		expect(component.find(".modals")).toHaveLength(1);
		component.unmount();

	});

	it("has 2 Buttons elements", () => {

		expect(component.find(Button)).toHaveLength(2);

	});

});
