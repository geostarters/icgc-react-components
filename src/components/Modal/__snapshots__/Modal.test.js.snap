// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component renders correct properties 1`] = `
<Modal
  actionbuttons={
    Array [
      Object {
        "color": "red",
        "handler": [MockFunction],
        "icon": "remove",
        "label": "No",
      },
      Object {
        "color": "green",
        "handler": [MockFunction],
        "icon": "check",
        "label": "Sí",
      },
    ]
  }
  centered={true}
  closeIcon={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  headertitle="Hola"
  open={true}
  trigger={null}
>
  <ModalHeader
    content="Hola"
  />
  <ModalContent />
  <ModalActions>
    <Button
      as="button"
      color="red"
      key="0"
      onClick={[MockFunction]}
      primary={false}
    >
      <Icon
        as="i"
        name="remove"
      />
       
      No
    </Button>
    <Button
      as="button"
      color="green"
      key="1"
      onClick={[MockFunction]}
      primary={false}
    >
      <Icon
        as="i"
        name="check"
      />
       
      Sí
    </Button>
  </ModalActions>
</Modal>
`;
